plugins/syncbeacon/plugin.go

```diff
@@ -0,0 +1,197 @@
+	})
+	return plugin
+}
+
+// configure events
+func configure(_ *node.Plugin) {
+	log = logger.NewLogger(PluginName)
+	beaconNodesPublicKeys = config.Node().GetStringSlice(CfgSyncBeaconFollowNodes)
+	beaconNodesStatusMap = make(map[string]Payload, len(beaconNodesPublicKeys))
+	for _, key := range beaconNodesPublicKeys {
+		beaconNodesStatusMap[key] = Payload{}
+	}
+	mgr = gossip.Manager()
+
+	gossip.Manager().Events().MessageReceived.Attach(events.NewClosure(func(event *gossipPkg.MessageReceivedEvent) {
+		payload := &Payload{}
+		err := payload.Unmarshal(event.Data)
+		if err != nil {
+			log.Info("error unmarshalling", err)
+		} else {
+			handlePayload(payload, event.Peer)
+		}
+	}))
+}
+
+// handlePayload handles the received payload. It does the following checks:
+// - The payload is a syncbeacon payload. Then it checks that the issuer of the payload is a followed node.
+// - The time that payload was sent is not greater than  CfgSyncBeaconMaxTimeWindowSec. If the duration is longer than CfgSyncBeaconMaxTimeWindowSec, we consider that beacon to be out of sync till we receive a newer payload.
+// - If the node is synced and more than syncPercentage of followed nodes are also synced, the node is set to synced. Otherwise, its set as desynced.
+func handlePayload(syncBeaconPayload *Payload, peer *peer.Peer) {
+	if !IsSyncBeaconPayload(syncBeaconPayload) {
+		return
+	}
+	peerPublicKey := peer.PublicKey().String()
+
+	//check if peer is in configured beacon follow list
+	shouldAccept := false
+	for i := 0; i < len(beaconNodesPublicKeys); i++ {
+		if beaconNodesPublicKeys[i] == peerPublicKey {
+			shouldAccept = true
+			break
+		}
+	}
+	if !shouldAccept {
+		return
+	}
+
+	dur := time.Since(time.Unix(0, syncBeaconPayload.sentTime))
+	if dur.Seconds() > float64(config.Node().GetInt(CfgSyncBeaconMaxTimeWindowSec)) {
+		log.Infof("syncbeacon payload received from %s is too old", peerPublicKey)
+		syncBeaconPayload.syncStatus = false
+	}
+
+	beaconNodesStatusMap[peerPublicKey] = *syncBeaconPayload
+	updateSynced()
+}
+
+// updateSynced checks the beacon nodes and update the nodes sync status
+func updateSynced() {
+	beaconNodesSyncedCount := 0.0
+	for _, p := range beaconNodesStatusMap {
+		if p.SyncStatus() {
+			beaconNodesSyncedCount++
+		}
+	}
+	isBeaconSynced := true
+	if len(beaconNodesStatusMap) > 0 {
+		isBeaconSynced = beaconNodesSyncedCount/float64(len(beaconNodesStatusMap)) >= syncPercentage
+	}
+	isLocalSynced := sync.Synced()
+	if isLocalSynced && isBeaconSynced {
+		sync.MarkSynced()
+	} else {
+		sync.MarkDesynced()
+	}
+}
+
+// broadcastSyncBeaconPayload broadcasts its sync status to its neighbors.
+func broadcastSyncBeaconPayload() {
+	syncBeaconPayload := NewSyncBeaconPayload(sync.Synced(), time.Now().UnixNano())
+	//msg := message.New(message.EmptyId, message.EmptyId, time.Now(), local.GetInstance().LocalIdentity().PublicKey(), 0, syncBeaconPayload, 0, ed25519.Signature{})
+	mgr.SendMessage(syncBeaconPayload.Bytes())
```

Comment:
This is conceptually wrong. If you don't write the beacon into the message Tangle, then, nodes following this node will not be able to solidify anything from this beacon, thus, their sync status would be meaningless.